jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/IntervalGraphRecognizer.java

```diff
@@ -0,0 +1,142 @@
+package org.jgrapht.alg.intervalgraph;
+
+import org.jgrapht.Graph;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+import static org.jgrapht.alg.intervalgraph.LexBreadthFirstSearch.*;
+
+public final class IntervalGraphRecognizer {
+
+  /**
+   * check if the given graph is an interval graph
+   *
+   * @param graph the graph to be checked
+   * @param <V>   the generic type representing vertices
+   * @param <E>   the generic type representing edges
+   * @return
+   */
+  public static <V, E> boolean isIntervalGraph(Graph<V, E> graph) {
```

Comment:
We have to special case the empty graph (which is an interval graph) and throw an exception if graph is null.